src/IO/AzureBlobStorage/PocoHTTPClient.cpp

```diff
@@ -0,0 +1,537 @@
+                UInt64 sleep_us = sleep_ns / 1000UL;
+                ProfileEvents::increment(ProfileEvents::AzureGetRequestThrottlerCount);
+                ProfileEvents::increment(ProfileEvents::AzureGetRequestThrottlerSleepMicroseconds, sleep_us);
+
+                if (for_disk_azure)
+                {
+                    ProfileEvents::increment(ProfileEvents::DiskAzureGetRequestThrottlerCount);
+                    ProfileEvents::increment(ProfileEvents::DiskAzureGetRequestThrottlerSleepMicroseconds, sleep_us);
+                }
+            }
+        }
+        else if (method == "PUT" || method == "POST" || method == "DELETE" || method == "PATCH")
+        {
+            if (put_request_throttler)
+            {
+                UInt64 sleep_ns = put_request_throttler->add(1);
+                UInt64 sleep_us = sleep_ns / 1000UL;
+                ProfileEvents::increment(ProfileEvents::AzurePutRequestThrottlerCount);
+                ProfileEvents::increment(ProfileEvents::AzurePutRequestThrottlerSleepMicroseconds, sleep_us);
+
+                if (for_disk_azure)
+                {
+                    ProfileEvents::increment(ProfileEvents::DiskAzurePutRequestThrottlerCount);
+                    ProfileEvents::increment(ProfileEvents::DiskAzurePutRequestThrottlerSleepMicroseconds, sleep_us);
+                }
+            }
+        }
+
+        Poco::URI uri;
+        uri.setScheme(url.GetScheme());
+        uri.setHost(url.GetHost());
+        uri.setPort(url.GetPort());
+        uri.setPath("/");
+
+        auto adaptive_timeouts = getTimeouts(method, first_attempt, true);
+
+        auto group = for_disk_azure ? HTTPConnectionGroupType::DISK : HTTPConnectionGroupType::STORAGE;
+        auto session = makeHTTPSession(
+            group,
+            uri,
+            adaptive_timeouts,
+            ProxyConfiguration{},
+            &connect_time
+        );
+
+        Stopwatch watch;
+        std::ostream & request_stream = session->sendRequest(poco_request, &connect_time, &first_byte_time);
+
+        addLatency(method, AzureLatencyType::Connect, connect_time);
+        addLatency(method, first_byte_latency_type, first_byte_time);
+        latency_recorded = true;
+
+        if (auto * body_stream = request.GetBodyStream(); body_stream != nullptr && body_stream->Length() > 0)
+        {
+            setTimeouts(*session, getTimeouts(method, first_attempt, /*first_byte*/ false));
+            body_stream->Rewind();
+
+            /// Manual copy
+            std::vector<uint8_t> buffer(8192);
+            while (auto read = body_stream->Read(buffer.data(), 8192))
+            {
+                if (read > 0)
+                    request_stream.write(reinterpret_cast<const char *>(buffer.data()), read);
+                else
+                    break;
+            }
+        }
+
+        // Set timeouts for receiving response
+        setTimeouts(*session, getTimeouts(method, first_attempt, /*first_byte=*/true));
```

Comment:
I do not see where we do 
`setTimeouts(*session, getTimeouts(method, first_attempt, /*first_byte=*/ __false__));`
for reading rest data from the response.